src/lib/Importer/SchemaImporter.php

```diff
@@ -175,6 +176,30 @@ private function addSchemaTableColumns(Table $table, array $columnList): void
                     $column->getName(),
                 );
             }
+
+            if (isset($columnConfiguration['foreignKey'])) {
+                $foreignKey = $columnConfiguration['foreignKey'];
+                $foreignField = $foreignKey['field'];
+
+                $table->addForeignKeyConstraint(
+                    $foreignKey['table'],
+                    [$columnName],
+                    [$foreignField],
+                    $foreignKey['options'] ?? [],
+                    $foreignKey['name'] ?? null,
+                );
+
+                if (isset($foreignKey['index'])) {
+                    $indexConfig = $this->normalizeIndexConfig($foreignKey['index'], $location);
+
+                    $this->addIndexToColumn(
+                        $indexConfig,
+                        $location . '.foreignKey',
+                        $table,
+                        $column->getName(),
+                    );
+                }
```

Comment:
could you extract this to a separate method to reduce cognitive complexity?

Reply:
Actually part of this code was unintentionally left (the part about `index`).

I initially intended to allow `foreignKey` to be able to have it's own `index` property, but then I realized that would would be always the same as `index` property on the field itself.